### ISSUE TITLE
Join and Else blocks aren't in the parser

### ISSUE THREAD

**Issue by mttrbrts** [accordproject/cicero#480]
**Describe the bug**
As reported in #479 by @JSv4.
Despite being in the documentation, the current 0.20 parser doesn't include tokens for `else` blocks and `join` blocks.

**To Reproduce**
Attempt to parse the attached template.

As a temporary measure, I'm going to remove these items from the documentation.

Rename to `elseblock.cta` [elseblock.zip](https://github.com/accordproject/cicero/files/3813618/elseblock.zip)

**Comment by jeromesimeon** [comment]
Glaring bug in the release. Thank @JSv4 for the report. I'm taking this one, unless you're already looking at it @mttrbrts ?

**Comment by jeromesimeon** [comment]
Looking into this. It seems the attached sample does not have an else block:
```
bash-3.2$ cicero parse 
17:28:28 - info: Using current directory as template folder
17:28:28 - info: Loading a default text/sample.md file.
17:28:30 - info:
{
  "$class": "org.accordproject.volumediscountlist.VolumeDiscountContract",
  "contractId": "124b6f65-fb5f-429c-b45f-d98e7d8d00c3",
  "isActive": true
}
bash-3.2$ cat text/grammar.tem.md 
{{#if isActive}}I'm active{{/if}}bash-3.2$ 
```

Here is one that shows the problem:

[elseblock.zip](https://github.com/accordproject/cicero/files/3817018/elseblock.zip)

```
bash-3.2$ cicero parse 
17:29:42 - info: Using current directory as template folder
17:29:42 - info: Loading a default text/sample.md file.
17:29:43 - error: invalid syntax at line 1 col 29:

  {{#if isActive}}I'm active{{else
                              ^
Unexpected varid token: "else"
```

**Comment by jeromesimeon** [comment]
With PR #481 : 
```
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js parse 
13:42:49 - info: Using current directory as template folder
13:42:49 - info: Loading a default text/sample.md file.
13:42:51 - info:
{
  "$class": "org.accordproject.volumediscountlist.VolumeDiscountContract",
  "contractId": "543cf793-834d-49a1-963b-aa7259f229b5",
  "isActive": true
}
```

Also works  on the else clause, and returns a parse error if neither text matches:
```
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js parse --sample text/sample-inactive.md 
13:43:55 - info: Using current directory as template folder
13:43:56 - info:
{
  "$class": "org.accordproject.volumediscountlist.VolumeDiscountContract",
  "contractId": "1ff3e329-cd97-4e31-80a6-dea67aa885b9",
  "isActive": true
}
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js parse --sample text/sample-err.md 
13:44:00 - info: Using current directory as template folder
13:44:01 - error: invalid syntax at line 1 col 5:

  I'm XXX active
      ^
Unexpected "X"

```

**Comment by jeromesimeon** [comment]
Here is a join block sample, added as part of the tests:
[block-join.zip](https://github.com/accordproject/cicero/files/3821198/block-join.zip)

Trace:
```
bash-3.2$ cat text/grammar.tem.md 
This is a list: {{#join rates ", "}}{{rate}}%{{/join}} (And more Text)
bash-3.2$ cat text/sample.md
This is a list: 3.1%, 3.3%, 2.9% (And more Text)
bash-3.2$ 
bash-3.2$ ~/git/cicero/packages/cicero-cli/index.js parse 
13:46:18 - info: Using current directory as template folder
13:46:18 - info: Loading a default text/sample.md file.
13:46:19 - info:
{
  "$class": "org.accordproject.volumediscountlist.VolumeDiscountContract",
  "contractId": "ceb30ded-7f4d-4bf9-a85b-f33d453332ac",
  "rates": [
    {
      "$class": "org.accordproject.volumediscountlist.RateRange",
      "rate": 3.1
    }, 
    {
      "$class": "org.accordproject.volumediscountlist.RateRange",
      "rate": 3.3
    }, 
    {
      "$class": "org.accordproject.volumediscountlist.RateRange",
      "rate": 2.9
    }
  ]
}
```

**Comment by jeromesimeon** [comment]
A fix is in! #481
